### PR TITLE
Add timestamp on backup files, C++11 version

### PR DESCRIPTION
**Pull request by arthurmco**
Implements changes called on #52, but using C++11
This PR is also a superset of #52, so commits there will appear here.

(I would add there, but Github did not let me...)

### CODE REVIEW

**Comment by lbgracioso** [comment]
LGTM.

**Review by lbgracioso — APPROVED** [review]

File: Dependencies.cmake [hunk 1]
```diff
@@ -33,7 +33,9 @@ function(cloysterhpc_setup_dependencies)
 
   if(NOT TARGET spdlog::spdlog)
     if (cloysterhpc_ENABLE_CONAN)
-      CPMFindPackage(NAME spdlog)
+      CPMFindPackage(NAME spdlog
+       OPTIONS
+        "SPDLOG_FMT_EXTERNAL ON")
```
Comment (viniciusferrao) [inline Dependencies.cmake:38]: Does that works when using Conan instead of CPM?

Comment (arthurmco) [reply Dependencies.cmake:38]: Probably not... but I used conan and it worked.

Comment (viniciusferrao) [reply Dependencies.cmake:38]: Well whatever. That may be an issue at some point but may be not. If not we gonna open a bug for this.

The root of the problem here is that CPM seems to be the way forward for Package Management, but there's some packages that aren't available yet. So we rely on Conan for it.

A bigger issue will be when we move to `std::format`... what if this PR breaks with it?

Comment (arthurmco) [reply Dependencies.cmake:38]: I will look if CPM supports adding extra parameters

Regarding `std::format`, it supports formatting chrono data types ([source](https://en.cppreference.com/w/cpp/utility/format/format)):

> For chrono types, the format specification is interpreted as [chrono format specification](https://en.cppreference.com/w/cpp/chrono/system_clock/formatter#Format_specification).

**Pull request merged**